ref/src/interpolate.f90

```diff
@@ -0,0 +1,160 @@
+module interpolate
+  implicit none
+
+contains
+
+  subroutine interpolate_allocate(odims, interpFactor, fdims)
+
+    ! Subroutine for calculating the lower and upper bounds of an
+    ! interpolated array with regard to interpFactor.  The user is
+    ! expected to call this routine, allocate the interpolated array,
+    ! then call subroutine: interpolate_arrray to fill the
+    ! interpolated array.
+    !
+    ! Subroutine parameters:
+    ! originalArray, 2D array to be interpolated
+    ! odims, 4 elements integer array, lower/upper subscripts of original array
+    ! interpFactor, interpolation factor
+    ! odims, 4 elements integer array, lower/upper subscripts of interpolated
+    ! array
+    !
+    ! For example:
+    ! dimension interpArray(:,:)
+    ! call interpolate_allocate(odims,interpFactor,fdims)
+    ! l1=fdims(1)
+    ! u1=fdims(2)
+    ! l2=fdims(3)
+    ! l2=fdims(4)
+    ! allocate (interpArray(l1:u1,l2:u2))
+    ! call interpolate_array(originalArray, odims, interpolatedArray, fdims,
+    ! interpFactor)
+
+    integer, intent(in) :: odims(4)
+    integer, intent(in) :: interpFactor
+    integer, intent(out):: fdims(4)
+
+    fdims(1) = odims(1)
+    fdims(3) = odims(3)
+    fdims(2) = odims(2) + (odims(2) - odims(1)) * interpFactor
+    fdims(4) = odims(4) + (odims(4) - odims(3)) * interpFactor
+
+  end subroutine interpolate_allocate
+
+  subroutine interpolate_array(oa, odims, f, fdims, interpFactor)
+
+    ! Bilinear interpolation of f with respect to oa.
```

Comment:
Consider moving the documentation of the interpolation to a comment block immediately above this routine so that the code is more easily read without large blocks of comments in between.